packages/php/blueprint/src/UsePluginHelpers.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Automattic\WooCommerce\Blueprint;
+
+trait UsePluginHelpers {
+	use UseWPFunctions;
+
+	public function activate_plugin_by_slug( $slug ) {
+		// Get all installed plugins
+		$all_plugins = $this->wp_get_plugins();
+
+		// Loop through all plugins to find the one with the specified slug
+		foreach ( $all_plugins as $plugin_path => $plugin_info ) {
+			// Check if the plugin path contains the slug
+			if ( strpos( $plugin_path, $slug . '/' ) === 0 ) {
+				// Deactivate the plugin
+				return $this->wp_activate_plugin( $plugin_path );
+			}
+		}
+		return false;
+	}
+
+	// Function to deactivate and delete a plugin by its slug
+	public function delete_plugin_by_slug( $slug ) {
+		// Get all installed plugins
+		$all_plugins = $this->wp_get_plugins();
+
+		// Loop through all plugins to find the one with the specified slug
+		foreach ( $all_plugins as $plugin_path => $plugin_info ) {
+			// Check if the plugin path contains the slug
+			if ( strpos( $plugin_path, $slug . '/' ) === 0 ) {
+				// Deactivate the plugin
+				$this->deactivate_plugin_by_slug( $slug );
```

Comment:
If `deactivate_plugin_by_slug` returns `false,` the method will still proceed to attempt to delete the plugin. Is this expected?

Reply:
@chihsuan That's a good question, and I think it's up for debate. The primary purpose of the function is to delete regardless of whether the plugin is activated. I think it's reasonable to attempt deletion even if the given plugin fails to deactivate. What do you think?

Reply:
@chihsuan Updated in [Plugin must be deactivated before attemping to delete it](https://github.com/woocommerce/woocommerce/pull/49763/commits/14ea8166430f85f7b06bb2845e8f33c80e5569dd)